Firestore/core/src/util/executor_std.cc

```diff
@@ -145,19 +164,19 @@ void ExecutorStd::Cancel(const Id operation_id) {
   }
 }
 
-ExecutorStd::Id ExecutorStd::PushOnSchedule(const TimePoint when,
-                                            const Tag tag,
-                                            Operation&& operation) {
+ExecutorStd::Id ExecutorStd::PushOnScheduleLocked(const TimePoint when,
+                                                  const Tag tag,
+                                                  Operation&& operation) {
   // Note: operations scheduled for immediate execution don't actually need an
   // id. This could be tweaked to reuse the same id for all such operations.
-  const auto id = NextId();
+  const auto id = NextIdLocked();
   state_->schedule_.Push(
       Task::Create(nullptr, when, tag, id, std::move(operation)));
   return id;
 }
 
-void ExecutorStd::PollingThread() {
-  std::shared_ptr<SharedState> local_state = state_;
+void ExecutorStd::PollingThread(std::shared_ptr<SharedState> state) {
+  std::shared_ptr<SharedState> local_state = state;
```

Comment:
How does the other copy of the shared pointer help? It seems they have exactly the same lifetime.

Reply:
It doesn't. I should have cleaned this up when changing this to pass the state rather than rely on `this`.